lib/datadog/tracing.rb

```diff
@@ -0,0 +1,118 @@
+module Datadog
+  # Datadog APM tracing public API.
+  #
+  # The Datadog teams ensures that public methods in this module
+  # only receive backwards compatible changes, and breaking changes
+  # will only occur in new major versions of `ddtrace`.
+  module Tracing
+    class << self
+      # TODO: copy docs from Datadog::Tracer#trace
+      def trace(*args, &block)
+        tracer.trace(*args, &block)
+      end
+      ruby2_keywords :trace if respond_to?(:ruby2_keywords, true)
+
+      # The currently active {Datadog::Tracer} instance.
+      #
+      # The instance returned can change throughout the lifetime of the application.
+      # This means it is not advisable to cache it.
+      #
+      # TODO: I think this next paragraph can be better written.
+      # Most of the functionality available through the {.tracer} instance is
+      # also available in public methods in the {Datadog::Tracing} module.
+      # It is preferable to use the public methods in the {Datadog::Tracing} when possible
+      # as {Datadog::Tracing} strongly defines the tracing public API, and thus
+      # we strive to no introduce breaking changes to {Datadog::Tracing} methods.
+      #
+      # @return [Datadog::Tracer] the active tracer
+      def tracer
+        components.tracer
+      end
```

Comment:
I'm curious, with the other API changes we're doing, what are the remaining use cases where people directly need to call methods on `tracer` that aren't available as methods on `Datadog::Tracing` directly?

Reply:
I don't know yet, I just started this task :)